src/Nethermind/Nethermind.TxPool/Filters/FeeTooLowFilter.cs

```diff
@@ -13,36 +13,72 @@ namespace Nethermind.TxPool.Filters
     /// <summary>
     /// Filters out transactions where gas fee properties were set too low or where the sender has not enough balance.
     /// </summary>
-    internal class FeeTooLowFilter : IIncomingTxFilter
+    internal sealed class FeeTooLowFilter : IIncomingTxFilter
     {
         private readonly IChainHeadSpecProvider _specProvider;
         private readonly IChainHeadInfoProvider _headInfo;
         private readonly TxDistinctSortedPool _txs;
         private readonly ILogger _logger;
 
-        public FeeTooLowFilter(IChainHeadInfoProvider headInfo, TxDistinctSortedPool txs, ILogManager logManager)
+        public FeeTooLowFilter(IChainHeadInfoProvider headInfo, TxDistinctSortedPool txs, ILogger logger)
         {
             _specProvider = headInfo.SpecProvider;
             _headInfo = headInfo;
             _txs = txs;
-            _logger = logManager.GetClassLogger();
+            _logger = logger;
+        }
+
+        private readonly static UInt256 GweiToWeiFactor = new UInt256(1_000_000_000);
+        private UInt256 GasLimitToWei(long gasLimitGwei)
+        {
+            return GweiToWeiFactor * new UInt256((ulong)gasLimitGwei);
         }
 
         public AcceptTxResult Accept(Transaction tx, TxFilteringState state, TxHandlingOptions handlingOptions)
         {
+            bool isTrace = _logger.IsTrace;
+            if (tx.GasLimit <= 0 || tx.GasLimit < Transaction.BaseTxGasCost)
+            {
+                // Not high enough GasLimit to run a txn
+                Metrics.PendingTransactionsTooLowFee++;
+                if (isTrace) _logger.Trace($"Skipped adding transaction {tx.ToString("  ")}, too low gas limit for a txn at price {tx.GasPrice}");
+                return AcceptTxResult.FeeTooLow.WithMessage("GasLimit too low to start txn at GasPrice");
+            }
+
             IReleaseSpec spec = _specProvider.GetCurrentHeadSpec();
-            UInt256 balance = state.SenderAccount.Balance;
-            UInt256 affordableGasPrice = tx.CalculateAffordableGasPrice(spec.IsEip1559Enabled, _headInfo.CurrentBaseFee, balance);
-            bool isNotLocal = (handlingOptions & TxHandlingOptions.PersistentBroadcast) != TxHandlingOptions.PersistentBroadcast;
+            bool isEip1559Enabled = spec.IsEip1559Enabled;
+            if (isEip1559Enabled && tx.IsEip1559)
+            {
+                if (tx.GasLimit < spec.MinGasLimit || tx.GasPrice < spec.Eip1559BaseFeeMinValue)
+                {
+                    // Amounts too low for spec
+                    Metrics.PendingTransactionsTooLowFee++;
+                    return AcceptTxResult.FeeTooLow.WithMessage("GasLimit or GasPrice too low for Eip1559 spec");
+                }
+            }
```

Comment:
`MinGasLimit` & `Eip1559BaseFeeMinValue` are block parameters, not transaction ones. Please remove this part.

Reply:
Removed